docs/inference-providers/guides/structured-output.md

```diff
@@ -0,0 +1,467 @@
+# Structured Outputs with Inference Providers
+
+In this guide, we'll show you how to use Inference Providers to generate structured outputs that follow a specific JSON schema. This is incredibly useful for building reliable AI applications that need predictable, parseable responses.
+
+Instead of hoping the model returns properly formatted JSON, structured outputs guarantee that the response will match your exact schema every time. This eliminates the need for complex parsing logic and makes your applications more robust.
+
+<Tip>
+
+This guide assumes you have a Hugging Face account. If you don't have one, you can create one for free at [huggingface.co](https://huggingface.co).
+
+</Tip>
+
+## What Are Structured Outputs?
+
+Structured outputs make sure that AI model responses always follow a specific structure, typically a JSON Schema. This means you get predictable, type-safe data that integrates easily with your systems. Instead of hoping the model returns valid JSON, you give the AI a strict template to follow, so you always get data in the format you expect.
+
+Traditionally, getting structured data from LLMs required prompt engineering (asking the model to "respond in JSON format"), post-processing and parsing the response, and sometimes retrying when parsing failed. This approach is unreliable and can lead to brittle applications.
+
+With structured outputs, you get:
+
+- Guaranteed compliance with your defined schema
+- Fewer errors from malformed or unparsable JSON
+- Easier integration with downstream systems
+- No need for retry logic or complex error handling
+- More efficient use of tokens (less verbose instructions)
+
+In short, structured outputs make your applications more robust and reliable by ensuring every response matches your schema, with built-in validation and type safety.
+
+## Step 1: Define Your Schema
+
+Before making any API calls, you need to define the structure you want. Let's build a practical example: extracting structured information from research papers. This is a common real-world use case where you need to parse academic papers and extract key details like title, authors, contributions, and methodology.
+
+We'll create a simple schema that captures the most essential elements: the paper's title and a summary of its abstract. The easiest way to do this is to use Pydantic, a library that allows you to define Python classes that represent JSON schemas (among other things).
+
+<hfoptions id="json-pydantic">
```

Comment:
I wouldn't use the toggles here since it sets up as a kind of "choose one or the other".  After showing the Pydantic code example, maybe transition with:

"The Pydantic represented JSON schema is shown below."